src/adapters/marketplace/_resolver.js

```diff
@@ -0,0 +1,284 @@
+import V00_MarkeplaceAdapter from './v00'
+import V01_MarkeplaceAdapter from './v01'
+import {
+  parseListingId,
+  parseOfferId,
+  generateListingId,
+  generateOfferId,
+  generateNotificationId
+} from '../../utils/id'
+import {
+  Notification,
+  readStatus,
+  unreadStatus,
+  storeKeys
+} from '../../models/notification'
+
+class MarketplaceResolver {
+  constructor({ contractService, ipfsService, store }) {
+    this.adapters = {
+      '000': new V00_MarkeplaceAdapter({ contractService, ipfsService, store }),
+      '001': new V01_MarkeplaceAdapter({ contractService, ipfsService, store })
+    }
+    this.versions = ['000', '001']
+    this.currentVersion = this.versions[this.versions.length - 1]
+    this.currentAdapter = this.adapters[this.currentVersion]
+    this.contractService = contractService
+    this.ipfsService = ipfsService
```

Comment:
It does not seem ipfsService is used anywhere in this class. Can we get rid of it ?